tests/sentry/integrations/test_notification_utilities.py

```diff
@@ -0,0 +1,68 @@
+from sentry.integrations.notifications import get_integrations_by_channel_by_recipient
+from sentry.models import Identity, IdentityProvider, IdentityStatus
+from sentry.testutils.cases import TestCase
+from sentry.testutils.helpers.notifications import DummyNotification
+from sentry.testutils.helpers.slack import install_slack
+from sentry.types.integrations import ExternalProviders
+
+
+class TestNotificationUtilities(TestCase):
+    def setUp(self):
+        super().setUp()
+        self.notification = DummyNotification(self.organization)
+
+        self.integration = install_slack(self.organization)
+
+        self.idp = IdentityProvider.objects.create(type="slack", external_id="TXXXXXXX1", config={})
```

Comment:
we have fixtures to help with setup. In this case `self.create_identity_provider`

https://github.com/getsentry/sentry/blob/master/src/sentry/testutils/fixtures.py#L391

You can do the same with identity

Reply:
This is so good, reduced a lot of boilerplate.